apps/web/app/(app)/onboarding/components/Product.tsx

```diff
@@ -73,7 +73,25 @@ const Product: React.FC<Product> = ({ done, isLoading, environmentId, product })
   if (!product) {
     return <ErrorComponent />;
   }
-
+  const buttonStyle = {
+    backgroundColor: color,
+    color: isColorNearWhite(color) ? "black" : "white",
+  };
+  function isColorNearWhite(color: string) {
+    const colorRegExp = /^#(..)(..)(..)$/;
+    const match = colorRegExp.exec(color);
+    if (match) {
+      const [, r, g, b] = match;
+      const red = parseInt(r, 16);
+      const green = parseInt(g, 16);
+      const blue = parseInt(b, 16);
+      const brightness = (0.299 * red + 0.587 * green + 0.114 * blue) / 255;
+      const threshold = 0.9;
+      return brightness > threshold;
+    } else {
+      return false;
+    }
+  }
```

Comment:
Hi @krishanu7, thanks a lot for your contribution 🎉 . We already have a helper method in the codebase called `isLight` at `apps/web/app/lib/utils.ts` location, which takes in a color and returns a `true` value if the color is a light color otherwise it returns false.

You can use the same method to check if the current selected color is light or dark, and then apply the relevant white or black text.
Could you please replace your logic and instead use the method we already have, I'll re-review it once you are done with the changes.

cc: @mattinannt